app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -86,6 +88,28 @@ def repos_needing_importer_updates(capsule, environment, content_view)
           end
         end
 
+        def remove_yum_clone_distributors(capsule)
+          repos_with_yum_clone_dist = []
+          repos = capsule.current_repositories_data
+          repos.each do |repo|
+            repo_details = capsule.pulp_repo_facts(repo["id"])
+            next unless repo_details
+            repo_details['distributors'].each do |distributor|
+              if distributor["distributor_type_id"] == "yum_clone_distributor"
+                repo_and_dist = [repo["id"], distributor["id"]]
+                repos_with_yum_clone_dist << repo_and_dist
+              end
+            end
+          end
+
+          repos_with_yum_clone_dist.each do |repo_id, dist_id|
+            plan_action(Pulp::Repository::DeleteDistributor, :repo_id => repo_id,
+                                                             :distributor_id => dist_id,
+                                                             :capsule_id => capsule.capsule.id
+                       )
+          end
+        end
```

Comment:
wouldn't it make more sense to remove any distributor that shouldn't be there rather than this specific one?

I.e. this should be caught in the  'distributors_match?' method and then in pulp/repository/refresh.rb add support for removing a distributor (there is already support for adding one).